internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -138,6 +141,11 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opt
 func (r *Reconciler) SetupForDryRun(recorder record.EventRecorder) {
 	r.desiredStateGenerator = desiredstate.NewGenerator(r.Client, r.ClusterCache, r.RuntimeClient)
 	r.recorder = recorder
+	r.externalTracker = external.ObjectTracker{
```

Comment:
Need special attention, I thought it does not have any other side effects so did this change make topology test pass.